pkg/cqrs/sqlitecqrs/sqlc/queries.sql

```diff
@@ -205,10 +205,33 @@ VALUES
 	(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);
 
 -- name: InsertTraceRun :exec
-INSERT OR REPLACE INTO trace_runs
-	(account_id, workspace_id, app_id, function_id, trace_id, run_id, queued_at, started_at, ended_at, status, source_id, trigger_ids, output, batch_id, is_debounce, cron_schedule)
-VALUES
-	(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);
+INSERT INTO trace_runs (
+    run_id, account_id, workspace_id, app_id, function_id, trace_id, 
+    queued_at, started_at, ended_at, status, source_id, trigger_ids, 
+    output, batch_id, is_debounce, cron_schedule, has_ai
+)
+VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+ON CONFLICT(run_id)
+DO UPDATE SET
+    account_id = excluded.account_id,
+    workspace_id = excluded.workspace_id,
+    app_id = excluded.app_id,
+    function_id = excluded.function_id,
+    trace_id = excluded.trace_id,
+    queued_at = excluded.queued_at,
+    started_at = excluded.started_at,
+    ended_at = excluded.ended_at,
+    status = excluded.status,
+    source_id = excluded.source_id,
+    trigger_ids = excluded.trigger_ids,
+    output = excluded.output,
+    batch_id = excluded.batch_id,
+    is_debounce = excluded.is_debounce,
+    cron_schedule = excluded.cron_schedule,
+    has_ai = CASE
```

Comment:
Once a run gets marked as having AI it cannot be undone. Subsequent non-ai steps are not allowed to set it back to false. I did this to avoid an extra fetch but am open to suggestions on how to do it differently if this is too ugly.

Reply:
This is fair. I think you can do the `CASE` inline though and keep the structure:
```sql
-- name: InsertTraceRun :exec
INSERT OR REPLACE INTO trace_runs
    (account_id, workspace_id, app_id, function_id, trace_id, run_id, queued_at, started_at, ended_at, status, source_id, trigger_ids, output, batch_id, is_debounce, cron_schedule, has_ai)
VALUES
    (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, 
        CASE
            WHEN ? = 1 THEN 1
            ELSE has_ai
        END
    );
```

Reply:
Discussed elsewhere but for the record. Inline case with named args does work but generates the wrong go type for has_ai:

```
   CASE
      WHEN sqlc.arg(has_ai) THEN 1 
      ELSE sqlc.arg(has_ai)
  END```
  
and for some reason...probably some sqlc + sqlite quirk...telling sqlc the type does not work:

```
CASE
        WHEN sqlc.arg(has_ai)::bool THEN 1 
        ELSE sqlc.arg(has_ai)::bool
 END```
 
 So, leaving it as it is above for now.